src/Bitcoin/Transaction.h

```diff
@@ -61,12 +61,23 @@ struct Transaction {
     Data getSequenceHash() const;
     Data getOutputsHash() const;
 
+    enum SegwitFormatMode {
+        NonSegwit,
+        IfHasWitness,
```

Comment:
What is different between `IfHasWitness` and `Segwit` ?

Reply:
The difference is in the case when there is no segwit scripts actually used.  With always Segwit setting, segwit format will be used in this case as well, meaning the 2 bytes of marker&flag, plus a 0 witness size will be included.  In the IfHasWitness case, in this case no segwit format is used, so w/o those 3 bytes.  Code used and continues to use this latter.